File: app/src/main/java/de/build_a_hero/app/SettingsActivity.java
package de.build_a_hero.app;

import android.os.Bundle;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.support.v4.app.Fragment;



import com.example.ninad.buildahero.ContactFragment;
import com.example.ninad.buildahero.CreditFragment;


public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        configureBackButton();

    }

    private void configureBackButton() {

        Button backButton = findViewById(R.id.backButton);
        backButton.setOnClickListener(new View.OnClickListener() {

                                          @Override
                                          public void onClick(View view) {
                                              finish();
                                          }
                                      }
        );
    }

    public void changeFragment(View view){
        Fragment fragment;

        if(view == findViewById(R.id.contactButton)){
            fragment = new ContactFragment();
            FragmentManager fm = getSupportFragmentManager();
            FragmentTransaction ft = fm.beginTransaction();
            ft.replace(R.id.settingFragment,fragment);
            ft.commit();
        }
        if(view == findViewById(R.id.creditButton)){
            fragment = new CreditFragment();
            FragmentManager fm = getSupportFragmentManager();
            FragmentTransaction ft = fm.beginTransaction();
            ft.replace(R.id.settingFragment,fragment);
            ft.commit();
        }
    }
}

File: app/src/main/java/de/build_a_hero/app/SettingsActivity.kt
package de.build_a_hero.app

import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.example.ninad.buildahero.ContactFragment
import com.example.ninad.buildahero.CreditFragment
import kotlinx.android.synthetic.main.activity_settings.*


class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
    }

    fun changeFragment(view: View) {
        var fragment: Fragment

        if (view === findViewById<View>(R.id.contactButton)) {
            fragment = ContactFragment()
            val fm = supportFragmentManager
            val ft = fm.beginTransaction()
            ft.replace(R.id.settingFragment, fragment)
            ft.commit()
            contactButton.setBackgroundColor(Color.rgb(224,217,217));
            creditButton.setBackgroundColor(Color.WHITE);
        }
        if (view === findViewById<View>(R.id.creditButton)) {
            fragment = CreditFragment()
            val fm = supportFragmentManager
            val ft = fm.beginTransaction()
            ft.replace(R.id.settingFragment, fragment)
            ft.commit()
            creditButton.setBackgroundColor(Color.rgb(224,217,217));
            contactButton.setBackgroundColor(Color.WHITE);
        }
    }
}